processors/base/helpers.py

```diff
@@ -69,14 +59,39 @@ def clean_list(raw_list):
     return cleaned_list
 
 
-def clean_dict(raw_dict):
-    """Remove falsy values from dict.
+def get_cleaned_identifiers(identifiers):
+    """Remove falsy values and bad identifiers.
     """
-    cleaned_dict = {}
-    for key, value in raw_dict.items():
-        if value:
-            cleaned_dict[key] = value
-    return cleaned_dict
+    PATTERNS = {
+        'actrn': r'^ACTRN',
+        'chictr': r'^ChiCTR',
+        'drks': r'^DRKS',
+        'euctr': r'^EUCTR',
+        'gsk': r'^GSK',
+        'irct': r'^IRCT',
+        'isrctn': r'^ISRCTN',
+        'jprn': r'^(C|JPRN|UMIN)',
+        'kct': r'^KCT',
+        'nct': r'^NCT',
+        'ntr': r'^NTR',
+        'pactr': r'^PACTR',
+        'per': r'^PER',
+        'rbr': r'^RBR',
+        'rpcec': r'^RPCEC',
+        'takeda': r'^TAKEDA',
+        'tctr': r'^TCTR',
+        'who': r'^U',
```

Comment:
Not so easy) Because there could be many special cases of thinks like zeros in NCT And then we will start to add secondary ids like industry ones (https://github.com/opentrials/opentrials/issues/329). So I just wanted to say - there could be I suppose some amount of logic working only on identifiers.